### PR TITLE
changing cComposite Chat to newer c++ standart

### PR DESCRIPTION
**Pull request by 12xx12**
upgraded to new C++ for loops and fixed errors in previous redition which made it possible to copy the object and cause errors.

Fixes: #5027

Todo:
- [ ] add move constructor

### CODE REVIEW

File: src/CompositeChat.cpp [hunk 1]
```diff
@@ -159,17 +159,19 @@ void cCompositeChat::ParseText(const AString & a_ParseText)
 
 			case ':':
 			{
-				const char * LinkPrefixes[] =
+				static const std::array<AString, 2> LinkPrefixes =
```
Comment (peterbell10) [inline src/CompositeChat.cpp:162]:
```suggestion
				static const std::array<std::string_view, 2> LinkPrefixes =
```

File: src/CompositeChat.cpp [hunk 2]
```diff
@@ -159,17 +159,19 @@ void cCompositeChat::ParseText(const AString & a_ParseText)
...
 				};
-				for (size_t Prefix = 0; Prefix < ARRAYCOUNT(LinkPrefixes); Prefix++)
+				for (const auto & Prefix : LinkPrefixes)
 				{
-					size_t PrefixLen = strlen(LinkPrefixes[Prefix]);
+					size_t PrefixLen = Prefix.size();
 					if (
 						(i >= first + PrefixLen) &&  // There is enough space in front of the colon for the prefix
-						(strncmp(a_ParseText.c_str() + i - PrefixLen, LinkPrefixes[Prefix], PrefixLen) == 0)  // the prefix matches
+						(strncmp(a_ParseText.c_str() + i - PrefixLen, Prefix.c_str(), PrefixLen) == 0)  // the prefix matches
```
Comment (peterbell10) [inline src/CompositeChat.cpp:174]:
```suggestion
						(std::string_view(a_ParseText).substr(i, PrefixLen) == Prefix)  // the prefix matches
```

File: src/CompositeChat.cpp [hunk 3]
```diff
@@ -45,10 +45,10 @@ cCompositeChat::~cCompositeChat()
...
 void cCompositeChat::Clear(void)
 {
-	for (cParts::iterator itr = m_Parts.begin(), end = m_Parts.end(); itr != end; ++itr)
+	for (auto & Part : m_Parts)
 	{
-		delete *itr;
-	}  // for itr - m_Parts[]
+		Part.reset();
+	}
 	m_Parts.clear();
```
Comment (peterbell10) [inline src/CompositeChat.cpp:52]: Don't need to explicitly reset the unique_ptrs.

Comment (12xx12) [reply src/CompositeChat.cpp:52]: hm, so we don't need a explicit destructor?

Comment (peterbell10) [reply src/CompositeChat.cpp:52]: `clear()` destroys all the elements automatically.

**Review by peterbell10 — APPROVED** [review]

**Comment by 12xx12** [comment]
waiting 'till tomorrow until merge

**Pull request merged**